Realm/RLMResults.mm

```diff
@@ -563,6 +571,138 @@ - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMResults *, RLMCollec
     return _results.add_notification_callback(RLMWrapCollectionChangeCallback(block, self, true), std::move(keyPaths));
 }
 
+- (void)completionWithThreadSafeReference:(RLMThreadSafeReference * _Nullable)reference
+                               confinedTo:(RLMScheduler *)confinement
+                               completion:(RLMResultsCompletionBlock)completion
+                                    error:(NSError *_Nullable)error {
+    auto tsr = (error != nil) ? nil : reference;
+    RLMRealmConfiguration *configuration = _realm.configuration;
+    [confinement invoke:^{
+        if (reference) {
+            RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
```

Comment:
This *should* never fail, but since we have the ability to properly report errors from here we might as well do so.